interp/interp_example_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package interp_test
+
+import (
+	"fmt"
+	"math"
+
+	"gonum.org/v1/gonum/interp"
+)
+
+func ExamplePredictor() {
+	xs := []float64{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}
+	ys := []float64{0, 0.001, 0.002, 0.1, 1, 2, 2.5, -10, -10.01, 2.49, 2.53, 2.55}
+
+	var pc interp.PiecewiseConstant
+	var pl interp.PiecewiseLinear
+	var as interp.AkimaSpline
+	var fb interp.FritschButland
+
+	_ = pc.Fit(xs, ys)
+	_ = pl.Fit(xs, ys)
+	_ = as.Fit(xs, ys)
+	_ = fb.Fit(xs, ys)
+
+	predictors := []interp.Predictor{&pc, &pl, &as, &fb}
+
+	n := len(xs)
+	dx := 0.25
+	nPts := int(math.Round(float64(n-1)/dx)) + 1
+
+	fmt.Printf("x\t\tPC\t\tPL\t\tAS\t\tFB\n")
+	for i := 0; i < nPts; i++ {
+		x := xs[0] + float64(i)*dx
+		fmt.Printf("%.2f", x)
+		for _, predictor := range predictors {
+			y := predictor.Predict(x)
+			fmt.Printf("\t%.2f", y)
+		}
+		fmt.Printf("\n")
+	}
+	// Output:
+	// x		PC		PL		AS		FB
```

Comment:
it's not really visually super appealing.
couldn't we find a better formatting strategy?
use [text/tabwriter](https://godoc.org/text/tabwriter) perhaps?

Reply:
On my Windows PC the output was correct using the current code. I'll check out tabwriter.

Reply:
It's not safe to assume that everyone has the same tabstops set, please use the tabwriter. It won't make much difference to the code here since it take tab separated lines as input.